framework/include/materials/DerivativeMaterialInterface.h

```diff
@@ -63,7 +64,7 @@ class DerivativeMaterialInterface : public T
    * Fetch a pointer to a material property if it exists, otherwise return null
    */
   template<typename U>
-  MaterialProperty<U> * getMaterialPropertyPointer(const std::string & name);
+  MaterialProperty<U> * getMaterialPropertyPointer(const std::string & name) __attribute__ ((deprecated));
```

Comment:
>  gives users a compile time deprecation warning. 

But only if the compiler supports that.  Otherwise users are not warned about using deprecated code.

> I find it rather useful. We had this in this file (for some methods I removed) and I think we had this in libmesh (in the Eigen contrib lib). I guess I don't see the downside...

The downside is inconsistency in the way functions are marked as deprecated within the framework.